File: src/main/java/javapair/Pair.java
package javapair;

public class Pair extends Object {
  public final int first;
  public final int second;

  public Pair(int fst, int snd) {
    first = fst;
    second = snd;
  }
  
  public String toString() {
  	return "Pair(" + first + ", " + second + ")";
  }
  
  public static Pair make(int fst, int snd) {
  	return new Pair(fst, snd);
  }
}

File: src/main/scala/scalapair/Pair.scala
package scalapair

class Pair(val first: Int, val second: Int) {
  def this() {
    this(0, 0)
  }
  
  def setFirst(fst: Int): Pair = new Pair(fst, second)
  
  override def toString(): String = s"Pair($first, $second)"
  
  def +(other: Pair): Pair = new Pair(first + other.first, second + other.second)
}

object Pair {
  def apply(fst: Int, snd: Int) = new Pair(fst, snd)
}